### PR TITLE
changes submodules to debians

### PR DESCRIPTION
**Pull request by xwilson03**
# PR Summary
PR Link: [Link](https://github.com/VEXU-GHOST/VEXU_GHOST/pull/76)

Issue Link: [Link](https://github.com/VEXU-GHOST/VEXU_GHOST/issues/75)

### Description
Switches compilation-heavy submodules for pre-compiled debians to assist CI pipeline and embedded devices.

### Reviewers
- Required: @MaxxWilson 

---
### Changelog
- Switches submodules to track VEXU_GHOST forks.
- Changes setup_submodules.sh script to pull debians from new ghost_dependencies repository instead of compiling from source locally.
- Skips debianized submodules in build and test scripts.
- Skips building visualization packages on arm machines (jetson).
- Adds reverse-ssh tunnel to CI pipeline for post-failure debugging.

### CODE REVIEW

File: scripts/setup_submodules.sh
Comment (MaxxWilson) [inline scripts/setup_submodules.sh]: usually we avoid adding hanging TODOs in PRs. Are you planning on doing this prior to merging, and if not, you should make an issue so that task is properly tracked.

File: scripts/setup_submodules.sh
Comment (MaxxWilson) [inline scripts/setup_submodules.sh:61]: This assumes that this variable is set. What happens if someone misses that step? Does the script exit, or does it continue and install in random places?

**Comment by MaxxWilson** [comment]
Looks pretty solid. Left relevant comments.

Another big question, do we know how this works for ROS Package dependencies? i.e. exporting the proper environment variables so that the ros packages are detected? None of these submodules are ros packages, and BehaviorTree related ones seem to take a bit.

I think should be good to merge AFTER you figure out the Jetson binaries or make an issue to do it later. Im approving for now because I dont have bandwidth to double check later, but I assume you will handle these before merging.

**Review by MaxxWilson — APPROVED** [review]

**Comment by MaxxWilson** [comment]
FOLLOWUP:

https://github.com/VEXU-GHOST/VEXU_GHOST/pull/77 This needs to get merged into this branch so we can debianize plotjuggler and plotjuggler_ros bc they are THICK and effect every build. We should also just do the cpp and lidar ones while we have this PR open, bc otherwise we never will lol.

- [ ] Debianize plotjuggler
- [ ] Debianize plotjuggler-ros
- [x] Debianize rplidar
- [x] Debianize BehaviorTreeROS2
- [x] Debianize BehaviorTreeCpp
- [ ] Add arm64 debians
- [x] Error checking on $VEXU_HOME

File: scripts/setup_submodules.sh
Comment (xwilson03) [reply scripts/setup_submodules.sh:61]: That gets caught on line 4; the script will exit after printing an error.

This variable is currently unused, but still necessary for future packages that may need to install their contents to the repository directory.

File: scripts/setup_submodules.sh
Comment (xwilson03) [reply scripts/setup_submodules.sh]: I can do that before this PR gets merged. This will be visited after the other dependencies get debianized.

**Comment by xwilson03** [comment]
> Looks pretty solid. Left relevant comments.
> 
> Another big question, do we know how this works for ROS Package dependencies? i.e. exporting the proper environment variables so that the ros packages are detected? None of these submodules are ros packages, and BehaviorTree related ones seem to take a bit.
> 
> I think should be good to merge AFTER you figure out the Jetson binaries or make an issue to do it later. Im approving for now because I dont have bandwidth to double check later, but I assume you will handle these before merging.

@MaxxWilson This should be possible through pre/post installation scripts.
Since the submodule debians are somewhat hand-crafted, maintainers would need to ensure that the proper environment variables get set post-install.

EDIT: do not use post install scripts in the way that i did! use bloom to handle this instead of trying to do it yourself: https://docs.ros.org/en/rolling/How-To-Guides/Building-a-Custom-Debian-Package.html

**Comment by xwilson03** [comment]
All x86 debians have been added, just not their arm equivalents.
Moving onto a small testing phase using the pipeline to ensure their quality; the build script will need to omit these packages going forward.

**Comment by xwilson03** [comment]
update on this; regular submodules continue to work flawlessly, but the mentioned approach to packaging ROS dependencies (postinst scripts that move files into the repository directory using $VEXU_HOME) was super hacky and now needs to be pretty much overhauled and replaced with bloom generated packages.

File: scripts/setup_submodules.sh
Comment (xwilson03) [reply scripts/setup_submodules.sh]: done! architecture is now checked against a list, which includes both amd64 and arm64. no hanging TODO's remaining, but the ghost_dependencies repository needs to be updated to contain those binaries.

**Comment by xwilson03** [comment]
tests for submodules got disabled since the submodules themselves werent technically "built"; some file location assertions were failing and necessitated this. either we need to skip submodule tests (current solution) or we figure out a way around this somehow.

other than that, though, pipeline is passing so this is ready for a merge. arm64 binaries are otw via the ghost_dependencies repo, but all the work here is done.

**Pull request merged**
